test/v2/controllers/CreateAmendHistoricNonFhlUkPropertyAnnualSubmissionControllerSpec.scala

```diff
@@ -0,0 +1,206 @@
+
+package v2.controllers
+
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.Result
+import uk.gov.hmrc.http.HeaderCarrier
+import v2.mocks.MockIdGenerator
+import v2.mocks.hateoas.MockHateoasFactory
+import v2.mocks.requestParsers.MockCreateAmendHistoricNonFhlUkPropertyAnnualSubmissionRequestParser
+import v2.mocks.services.{MockCreateAmendHistoricNonFhlUkPropertyAnnualSubmissionService, MockEnrolmentsAuthService, MockMtdIdLookupService}
+import v2.models.domain.{Nino, TaxYear}
+import v2.models.errors._
+import v2.models.hateoas.Method.GET
+import v2.models.hateoas.{HateoasWrapper, Link}
+import v2.models.outcomes.ResponseWrapper
+import v2.models.request.createAmendHistoricNonFhlUkPropertyAnnualSubmission.{CreateAmendHistoricNonFhlUkPropertyAnnualSubmissionFixture,
+  CreateAmendHistoricNonFhlUkPropertyAnnualSubmissionRawData, CreateAmendHistoricNonFhlUkPropertyAnnualSubmissionRequest}
+import v2.models.response.createAmendHistoricNonFhlUkPropertyAnnualSubmission.{CreateAmendHistoricNonFhlUkPropertyAnnualSubmissionHateoasData,
+  CreateAmendHistoricNonFhlUkPropertyAnnualSubmissionResponse}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class CreateAmendHistoricNonFhlUkPropertyAnnualSubmissionControllerSpec
+    extends ControllerBaseSpec
+    with MockEnrolmentsAuthService
+    with MockMtdIdLookupService
+    with MockCreateAmendHistoricNonFhlUkPropertyAnnualSubmissionService
+    with MockCreateAmendHistoricNonFhlUkPropertyAnnualSubmissionRequestParser
+    with MockHateoasFactory
+    with MockIdGenerator
+    with CreateAmendHistoricNonFhlUkPropertyAnnualSubmissionFixture {
+
+  private val nino          = "AA123456A"
+  private val taxYear       = "2022-23"
+  private val correlationId = "X-123"
+
+  trait Test {
+    val hc: HeaderCarrier = HeaderCarrier()
+
+    val controller = new CreateAmendHistoricNonFhlUkPropertyAnnualSubmissionController(
+      authService = mockEnrolmentsAuthService,
+      lookupService = mockMtdIdLookupService,
+      parser = mockCreateAmendHistoricNonFhlUkPropertyAnnualSubmissionRequestParser,
+      service = mockCreateAmendHistoricService,
+      hateoasFactory = mockHateoasFactory,
+      cc = cc,
+      idGenerator = mockIdGenerator
+    )
+
+    MockMtdIdLookupService.lookup(nino).returns(Future.successful(Right("test-mtd-id")))
+    MockedEnrolmentsAuthService.authoriseUser()
+    MockIdGenerator.getCorrelationId.returns(correlationId)
+  }
+
+  private val testHateoasLink = Link(href = s"/individuals/business/property/annual/$nino/annual/$taxYear", method = GET, rel = "self")
+
+  val hateoasResponse: JsValue = Json.parse(s"""
+       |{
+       |   "links": [
+       |      {
+       |         "href": "/individuals/business/property/annual/$nino/annual/$taxYear",
+       |         "method": "GET",
+       |         "rel": "self"
+       |      }
+       |   ]
+       |}
+    """.stripMargin)
+
+  private val requestJson = Json.parse(
```

Comment:
There's a fixture that this could use, in `CreateAmendHistoricNonFhlUkPropertyAnnualSubmissionFixture`.
Trouble is, that's in `v2.models.request`, i.e. the wrong place, obvs should be under fixtures. Perhaps that could be fixed at the same time? :sweat_smile: